src/tape/ensemble.py

```diff
@@ -37,8 +52,16 @@ def __init__(self, client=True, **kwargs):
         self._source = None  # Source Table
         self._object = None  # Object Table
 
-        self._source_dirty = False  # Source Dirty Flag
-        self._object_dirty = False  # Object Dirty Flag
+        self.frames = {}  # Frames managed by this Ensemble, keyed by label
+
+        # A unique ID to allocate new result frame labels.
+        self.default_frame_id = 1
+
+        self.source = None  # Source Table EnsembleFrame
+        self.object = None  # Object Table EnsembleFrame
+
+        self._source_temp = []  # List of temporary columns in Source
```

Comment:
potential duplicates

Reply:
Thanks for the catch, removed duplicate self._source_temp and self._object_temp